src/Compiler/Lambda/OptLambda.sml

```diff
@@ -3337,7 +3362,8 @@ structure OptLambda: OPT_LAMBDA =
 	let val (lamb,let_env) = functionalise_let let_env lamb
 	    val lamb = fix_conversion lamb
 	    val (lamb,inveta_env) = inverse_eta_for_fix_bound_lvars inveta_env lamb
-            val lamb = table2d_simplify lamb
+        val lamb = table2d_simplify lamb
+        val lamb = alloc_vectors lamb
```

Comment:
I see what you're trying to do but I think it would be easier if you just assume that storage is already allocated and that you implement a `f256_assign` function instead that will store four 64bit floats into already allocated memory, starting at an offset. Such a function can be used for storing into mono real-arrays and for creating a box-function that stores into scratch-memory...